Include docs on building IB/RDMA support (#584)
rapidsai/ucx-py: docs/source/install.rst

The following instructions assume you'll be using ucx-py on a CUDA enabled system and is in a `Conda environment <https://docs.conda.io/projects/conda/en/latest/>`_.

.. note::

>     As of version 0.15, the UCX conda package will no longer build with IB/RDMA included.  This is largely due in part to compatibility issues

Reviewer: ```suggestion
    As of version 0.15, the UCX conda package build will no longer include IB/RDMA support.  This is largely due in part to compatibility issues
```

    git clone https://github.com/openucx/ucx
    cd ucx
    git checkout v1.8.x

>     # apply UCX IB registration patches

Reviewer: ```suggestion
    # apply UCX IB registration cache patches, improves overall
    # IB performance when using a memory pool
```

>     curl -L https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch > /tmp/add-page-alignment.patch
>     curl -L https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch > /tmp/ib_registration_cache.patch

Reviewer 2: Could just do `curl -LO` to download them to a file of the same name in the current working directory.

```suggestion
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch
```

>     curl -L https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch > /tmp/add-page-alignment.patch
>     curl -L https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch > /tmp/ib_registration_cache.patch
>     git apply /tmp/ib_registration_cache.patch && git apply /tmp/add-page-alignment.patch

Reviewer 2: Could just do `curl -LO` to download them to a file of the same name in the current working directory.

```suggestion
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch
    git apply ib_registration_cache.patch && git apply add-page-alignment.patch
```

    ../contrib/configure-devel --prefix=$CONDA_PREFIX --with-cuda=$CUDA_HOME --enable-mt CPPFLAGS="-I/$CUDA_HOME/include"
    make -j install

UCX + OFED
~~~~~~~~~~

As noted above, the UCX conda package no longer builds support for IB/RDMA.  To build UCX with IB/RDMA support first confirm OFED is installed properly:

::

    (ucx) user@dgx:~$ ofed_info -s
    OFED-internal-4.7-3.2.9

If OFED drivers are not installed on the machine, you can download drivers at directly from `Mellanox <https://www.mellanox.com/products/infiniband-drivers/linux/mlnx_ofed>`_.  For versions older than 5.1 click on, *archive versions*.

> To build UCX with IB/RDMA support, include the ``--with-rdmacm``, ``--with-verbs``, and ``--with-cm`` build flags.  For example:

Reviewer: ```suggestion
To build UCX with IB/RDMA support, include the ``--with-rdmacm`` and ``--with-verbs`` build flags.  For example:
```

Reviewer: `cm` is deprecated, according to a comment by Yossi in Slack:

> there are several things called "cm" in different contexts, in particular usage of libibcm package and --with-cm config param are considered deprecated and will probably be removed

::
     ../contrib/configure-release --enable-mt
    --prefix="$CONDA_PREFIX" \
    --with-cuda="$CUDA_HOME" \
    --enable-mt \

>     --with-cm \

Reviewer: ```suggestion
```

    pip install .
    # or for develop build
    pip install -v -e .

> 
> 
> ##
> 
> 
> 
> (ucx) bzaitlen@dgx12:/datasets/bzaitlen/GitRepos/ucx$ git apply /tmp/ib_registration_cache.patch
> (ucx) bzaitlen@dgx12:/datasets/bzaitlen/GitRepos/ucx$ git apply /tmp/add-page-alignment.patch

Reviewer 2: Is this mean to still be here or can this be dropped?

```suggestion
```
